test/tests/dgkernels/advection_diffusion_mixed_bcs_test_resid_jac/dg_advection_diffusion_test.i

```diff
@@ -0,0 +1,105 @@
+[Mesh]
+  type = GeneratedMesh
+  nx = 2
+  dim = 1
+[]
+
+[Kernels]
+  [./source]
+    type = UserForcingFunction
+    variable = u
+    function = 'forcing_func'
+  [../]
+  [./convection]
+    type = ConservativeAdvection
+    variable = u
+    velocity = '1 0 0'
+  [../]
+  [./diffusion]
+    type = MatDiffusion
+    variable = u
+    prop_name = 'k'
+  [../]
+[]
+
+[DGKernels]
+  [./convection]
+    type = DGConvection
+    variable = u
+    velocity = '1 0 0'
+  [../]
+  [./diffusion]
+    type = DGDiffusion
+    variable = u
+    diff = 'k'
+    sigma = 6
+    epsilon = -1
+  [../]
+[]
+
+[BCs]
+  [./advection]
+    type = OutflowBC
+    boundary = 'right'
+    variable = u
+    velocity = '1 0 0'
+  [../]
+  [./diffusion_left]
+    type = DGFunctionDiffusionDirichletBC
+    boundary = 'left'
+    variable = u
+    sigma = 6
+    epsilon = -1
+    function = 'boundary_left_func'
+    diff = 'k'
+  [../]
+[]
+
+[Variables]
+  [./u]
+    family = MONOMIAL
```

Comment:
It is a libMesh thing, i do not know where it is exactly documented. It is basically broken LAGRANGE. Every element has its shape functions and they are directly used as the basis functions. They are node based, but can be discontinuous.